### PR TITLE
Initial Import of the yeoman generator

### PR DESCRIPTION
**Pull request by benoitf**
Allow to use yeoman tool to scaffold a Theia plug-in

CQs dependencies
- [x] yeoman-generator v3.0.0 
- [x] yeoman-generator v2.0.5 https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16842
- [x] yosay v2.0.2 https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16843

CQs dev-dependencies
- [x] rimraf v2.6.2 https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16844

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,23 @@
+# Theia plugin yeoman generator
+A [Yeoman](yeoman.io) generator that scaffolds a project structure for developing extensions to the [Theia IDE](https://github.com/theia-ide/theia).
 
+To use it, install yo and then generator
+
+```
+npm install -g yo @wiptheia/generator-theia-plugin
+```
```
Comment (evidolob) [inline README.md]: Do we still using `wiptheia` ?

File: src/app/index.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+            }]);
+            (this.options as any).pluginName = answers.name
+        }
+
+        if (!(this.options as any).pluginType) {
+
+            const answers = await this.prompt([{
+                type: 'list',
+                name: 'pluginType',
+                message: 'What type of extension do you want?',
```
Comment (evidolob) [inline src/app/index.ts]: `extension` or `plugin` ?

File: src/app/index.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                choices: [
+                    {
+                        name: 'Backend plugin, it will run on the server side.',
+                        value: 'backend'
+                    },
+                    {
+                        name: 'Frontend extension, it will run on the browser side.',
```
Comment (evidolob) [inline src/app/index.ts]: 'Frontend extension' or 'Frontend plugin'?

**Review by evidolob — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (benoitf) [reply README.md]: fixed

File: src/app/index.ts (hunk 2 above)
Comment (benoitf) [reply src/app/index.ts]: fixed

File: src/app/index.ts (hunk 3 above)
Comment (benoitf) [reply src/app/index.ts]: fixed

**Pull request merged**